### PR TITLE
Azure powershell SDK changes for Notification Hubs. Added a skuTier parameter for create/update namespace.

### PR DESCRIPTION
**Pull request by amolr**
## Description

The create/update namespace operation for Notification Hubs can now use a SkuTier parameter which sets the Sku level at the namespace level. This change allows the user to set the Sku using the parameter which creating or updating the namespace. In case of create, the skuTier if not specified is free by default.

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270%28v=vs.85%29.aspx).
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the Powershell CmdLet Help Editor
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**
### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).
### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@amolr__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (amolr). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

**Comment by shahabhijeet** [comment]
@amolr can you address failures

**Comment by cormacpayne** [comment]
@amolr Hey Amol, can you add the following tests:
- Using `New-AzureRmNotificationHubsNamespace` where the `SkuTier` is "Free" or isn't used at all
- Using `Set-AzureRmNotificationHubsNamespace` where the `SkuTier` is a value other than "Free"

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs/Commands/Namespace/NewAzureNotificationHubsNamespace.cs [hunk 1]
```diff
@@ -49,12 +49,23 @@ public class NewAzureNotificationHubsNamespace : AzureNotificationHubsCmdletBase
             HelpMessage = "Hashtables which represents resource Tags.")]
         public Hashtable Tags { get; set; }
 
+        [Parameter(Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            Position = 4,
+            HelpMessage = "Sku tier of the namespace")]
+        public string skuTier { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs/Commands/Namespace/NewAzureNotificationHubsNamespace.cs]: @amolr please rename this to `SkuTier`

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs/Commands/Namespace/SetAzureNotificationHubsNamespace.cs [hunk 2]
```diff
@@ -61,6 +61,12 @@ public class SetAzureNotificationHubsNamespace : AzureNotificationHubsCmdletBase
             HelpMessage = "Hashtables which represents resource Tags.")]
         public Hashtable Tags { get; set; }
 
+        [Parameter(Mandatory = false,
+          ValueFromPipelineByPropertyName = true,
+          Position = 4,
+          HelpMessage = "Sku tier of the namespace")]
+        public string skuTier { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs/Commands/Namespace/SetAzureNotificationHubsNamespace.cs]: @amolr please rename this to `SkuTier`

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs/Commands/Namespace/SetAzureNotificationHubsNamespace.cs [hunk 3]
```diff
@@ -78,7 +84,7 @@ public override void ExecuteCmdlet()
                 () =>
                 {
                     // Update a namespace 
-                    var nsAttribute = Client.UpdateNamespace(ResourceGroup, Namespace, Location, State, Critical, ConvertTagsToDictionary(Tags));
+                    var nsAttribute = Client.UpdateNamespace(ResourceGroup, Namespace, Location, State, Critical, ConvertTagsToDictionary(Tags), skuTier);
```
Comment (cormacpayne) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs/Commands/Namespace/SetAzureNotificationHubsNamespace.cs]: @amolr does `SkuTier` need to have a default value here like it does in `New-AzureNotificationHubsNamespace`? For example, if someone passes in an empty string for `SkuTier`, does it need to be set to "free"?

Comment (amolr) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs/Commands/Namespace/SetAzureNotificationHubsNamespace.cs]: No it need not have it here

File: src/ResourceManager/NotificationHubs/Commands.NotificationHubs.Test/Commands.NotificationHubs.Test.csproj [hunk 4]
```diff
@@ -71,15 +71,16 @@
...
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.NotificationHubs.2.0.0-preview\lib\net45\Microsoft.Azure.Management.NotificationHubs.dll</HintPath>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.NotificationHubs.2.1.0-preview\lib\net45\Microsoft.Azure.Management.NotificationHubs.dll</HintPath>
       <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.Azure.ResourceManager, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.Resources.2.20.0-preview\lib\net40\Microsoft.Azure.ResourceManager.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.Azure.Test.Framework">
+    <Reference Include="Microsoft.Azure.Test.Framework, Version=1.0.6052.28118, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (amolr) [inline src/ResourceManager/NotificationHubs/Commands.NotificationHubs.Test/Commands.NotificationHubs.Test.csproj]: I think this change got introduced while trying to debug an issue with Abhijeet related to our tests failing

**Comment by cormacpayne** [comment]
@amolr please update the [change log](https://github.com/Azure/azure-powershell/blob/dev/src/ResourceManager/NotificationHubs/ChangeLog.md) with the changes you have made in this PR

**Comment by markcowl** [comment]
@amolr Please update the changelog as Cormac suggested, here: https://github.com/Azure/azure-powershell/blob/dev/src/ResourceManager/NotificationHubs/ChangeLog.md

**Comment by markcowl** [comment]
@amolr still looking for commit cleanup

**Comment by cormacpayne** [comment]
LGTM :shipit:

**Pull request merged**